pkg/webhook/kapiserver/ensurer.go

```diff
@@ -163,3 +162,58 @@ func ensureKubeAPIServerIsMutated(ps *corev1.PodSpec, c *corev1.Container, caBun
 		},
 	})
 }
+
+// getLatestIssuedCABundleSecret returns the oidc-webhook latest CA bundle secret
+func getLatestIssuedCABundleSecret(ctx context.Context, c client.Client, namespace string) (*corev1.Secret, error) {
+	secretList := &corev1.SecretList{}
+	if err := c.List(ctx, secretList, client.InNamespace(namespace), client.MatchingLabels{
+		secretsmanager.LabelKeyBundleFor:       secrets.CAName,
+		secretsmanager.LabelKeyManagedBy:       secretsmanager.LabelValueSecretsManager,
+		secretsmanager.LabelKeyManagerIdentity: secrets.ManagerIdentity,
+	}); err != nil {
+		return nil, err
+	}
+	return getLatestIssuedSecret(secretList.Items)
+}
+
+// getLatestIssuedSecret returns the secret with the "issued-at-time" label that represents the latest point in time
+func getLatestIssuedSecret(secrets []corev1.Secret) (*corev1.Secret, error) {
+	if len(secrets) == 0 {
+		return nil, &NoCASecretError{}
+	}
+
+	newestSecret := &secrets[0]
+	initialIssuedAt, ok := secrets[0].Labels[secretsmanager.LabelKeyIssuedAtTime]
+	if !ok {
+		return nil, &NoIssuedAtTimeError{secretName: secrets[0].Name, namespace: secrets[0].Namespace}
+	}
+
+	initialIssuedAtUnix, err := strconv.ParseInt(initialIssuedAt, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	currentIssuedAtTime := time.Unix(initialIssuedAtUnix, 0).UTC()
```

Comment:
Why is this needed? The loop below has the same code?

Reply:
This initializes the initial "max" element of the slice that will be used for comparison in the loop below. I guess that there are some other ways to write this but I thought that this will be easy enough to read.

Reply:
OK, I don't request changes since it's your component/you are the owner, I was just wondering. Otherwise, this looks good to me.

Reply:
I will extract it into a function so that it is not repeated

Reply:
You could simply initialize variables and during the loop check if they are still unitialized. If yes, then the current secret can be stored (similar to https://github.com/gardener/gardener/blob/242a08c7eab6bd310b58d5031df8307fc8f96a6e/pkg/utils/secrets/manager/generate.go#L332-L342)